### PR TITLE
libnetwork: remove redundant init(), remove dead code, and fix stubs / build-tags

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to / some bits extracted from https://github.com/moby/moby/pull/43506

### libnetwork: fix stubs

- sandbox, endpoint changed in https://github.com/moby/moby/commit/c71555f03042ca531e24f6368d7e8c128774926f (https://github.com/moby/moby/pull/44805), but  missed updating the stubs.
- add missing stub for `Controller.cleanupServiceDiscovery()`
- While at it also doing some minor (formatting) changes.

### libnetwork: remove dead code on Windows

These methods are only called on unix.

### libnetwork/osl: rename var that collided with import

Also renaming another var for consistency ':-)

### libnetwork/osl: unify stubs for IfaceOption

Use the same signature for all platforms, but stub the nwIface type.

### libnetwork/osl: unify stubs for NeighOption

Use the same signature for all platforms, but stub the neigh type.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: libnetwork/osl/sandbox.go [hunk 1]
```diff
@@ -17,6 +17,12 @@ const (
 	SandboxTypeLoadBalancer = iota
 )
 
+// IfaceOption is a function option type to set interface options.
+type IfaceOption func(i *nwIface)
+
+// NeighOption is a function option type to set neighbor options.
+type NeighOption func(nh *neigh)
```
Comment (thaJeztah) [inline libnetwork/osl/sandbox.go:24]: TBH; we should look at these as they use a non-exported type as part of their type, which effectively means they cannot be implemented outside of this package.

They're currently part of an interface, but perhaps we don't need that if we move code to be with the correct build-tags (I kept that for a later exercise)

**Comment by thaJeztah** [comment]
@vvoland @akerouanton PTAL 🤗

**Review by vvoland — APPROVED** [review]

**Review by akerouanton — APPROVED** [review]
LGTM

**Comment by thaJeztah** [comment]
Thanks! Failures are `TestPushToCentralRegistryUnauthorized`, which is unrelated (and it looks like a new deploy went to Docker Hub, so may be resolved now)

**Pull request merged**